marketplaceItems/appreciate/appResources/appData/appreciate_app.js

```diff
@@ -0,0 +1,947 @@
+                                    "roughness": 0.0,
+                                    "albedo": [color.red/255, color.green/255, color.blue/255],
+                                }
+                            ]
+                        })
+                    });
+                }
+
+                Entities.editEntity(intensityEntity, {
+                    position: halfwayBetweenHands(),
+                    dimensions: Vec3.multiply(INTENSITY_ENTITY_MAX_DIMENSIONS, currentIntensity),
+                    angularVelocity: Vec3.multiply(currentInitialAngularVelocity,
+                        currentIntensity * ANGVEL_ENTITY_MULTIPLY_FACTOR),
+                    color: color
+                });
+            } else {
+                var props = INTENSITY_ENTITY_PROPERTIES;
+                props.position = halfwayBetweenHands();
+
+                currentInitialAngularVelocity.x = Math.random() *
+                    (INTENSITY_ENTITY_MAX_ANGULAR_VELOCITY.x -
+                    INTENSITY_ENTITY_MIN_ANGULAR_VELOCITY.x) + INTENSITY_ENTITY_MIN_ANGULAR_VELOCITY.x;
+                currentInitialAngularVelocity.y = Math.random() *
+                    (INTENSITY_ENTITY_MAX_ANGULAR_VELOCITY.y -
+                    INTENSITY_ENTITY_MIN_ANGULAR_VELOCITY.y) + INTENSITY_ENTITY_MIN_ANGULAR_VELOCITY.y;
+                currentInitialAngularVelocity.z = Math.random() *
+                    (INTENSITY_ENTITY_MAX_ANGULAR_VELOCITY.z -
+                    INTENSITY_ENTITY_MIN_ANGULAR_VELOCITY.z) + INTENSITY_ENTITY_MIN_ANGULAR_VELOCITY.z;
+                props.angularVelocity = currentInitialAngularVelocity;
+
+                intensityEntity = Entities.addEntity(props, "avatar");
+                
+                var materialProps = INTENSITY_MATERIAL_ENTITY_PROPS;
+                materialProps.parentID = intensityEntity;
+                intensityMaterialEntity = Entities.addEntity(materialProps, "avatar");
+            }
+        } else {
+            if (intensityEntity) {
+                Entities.deleteEntity(intensityEntity);
+                intensityEntity = false;
+            }
+            if (intensityMaterialEntity) {
+                Entities.deleteEntity(intensityMaterialEntity);
+                intensityMaterialEntity = false;
+            }
+            
+            maybeClearUpdateIntensityEntityInterval();
+        }
+    }
+
+    // Function that AppUI calls when the App's UI opens
+    function onOpened() {
+        updateCurrentIntensityUI();
+    }
+
+    // Locally pre-caches all of the sounds in the sounds/claps and sounds/whistles
+    // directories.
+    var NUM_CLAP_SOUNDS = 16;
+    var NUM_WHISTLE_SOUNDS = 17;
+    var clapSounds = [];
+    var whistleSounds = [];
+    function getSounds() {
+        for (var i = 1; i < NUM_CLAP_SOUNDS + 1; i++) {
+            clapSounds.push(SoundCache.getSound(Script.resolvePath(
+                "resources/sounds/claps/" + ("0" + i).slice(-2) + ".wav")));
```

Comment:
What does the .slice do in this case?  Don't you get the same result without it?  var file = "resources/sounds/claps/" + ("0" + 0).slice(-2) + ".wav"
console.log(file);

var file = "resources/sounds/claps/" + ("0" + 0) + ".wav"
console.log(file);

Reply:
ahh nevermind, sorry just realized that was to handle the double digit files

Reply:
np! indeed that's what it's for.